src/app/rundown-view/components/rundown/rundown.component.spec.ts

```diff
@@ -10,11 +10,210 @@ import { Logger } from '../../../core/abstractions/logger.service'
+      mockedRundownEventObserver: mockedRundownEventObserver,
+      mockedMiniShelfStateService: mockedMiniShelfStateService,
+      mockedRundownTimingContextStateService: mockedRundownTimingContextStateService,
+      mockedActionStateService: mockedActionStateService,
+    })
+
+    verify(mockedRundownEventObserver.subscribeToRundownDeactivation(anything())).once()
+  })
+
+  it('should subscribe to RundownAutoNext changes upon init', async () => {
+    const mockedMiniShelfStateService: MiniShelfStateService = createMockOfMiniShelfStateService()
+    const mockedRundownTimingContextStateService: RundownTimingContextStateService = createMockOfRundownTimingContextStateService()
+    const mockedActionStateService: ActionStateService = createMockOfActionStateService()
+    const mockedRundownEventObserver: RundownEventObserver = createMockOfRundownEventObserver()
+    const mockedRundown: Rundown = new TestEntityFactory().createRundown()
+
+    await configureTestBed({
+      mockedRundown: mockedRundown,
+      mockedRundownEventObserver: mockedRundownEventObserver,
+      mockedMiniShelfStateService: mockedMiniShelfStateService,
+      mockedRundownTimingContextStateService: mockedRundownTimingContextStateService,
+      mockedActionStateService: mockedActionStateService,
+    })
+
+    verify(mockedRundownEventObserver.subscribeToRundownAutoNext(anything())).once()
+  })
+
+  it('should subscribe to RundownSetNext changes upon init', async () => {
+    const mockedMiniShelfStateService: MiniShelfStateService = createMockOfMiniShelfStateService()
+    const mockedRundownTimingContextStateService: RundownTimingContextStateService = createMockOfRundownTimingContextStateService()
+    const mockedActionStateService: ActionStateService = createMockOfActionStateService()
+    const mockedRundownEventObserver: RundownEventObserver = createMockOfRundownEventObserver()
+    const mockedRundown: Rundown = new TestEntityFactory().createRundown()
+
+    await configureTestBed({
+      mockedRundown: mockedRundown,
+      mockedRundownEventObserver: mockedRundownEventObserver,
+      mockedMiniShelfStateService: mockedMiniShelfStateService,
+      mockedRundownTimingContextStateService: mockedRundownTimingContextStateService,
+      mockedActionStateService: mockedActionStateService,
+    })
+
+    verify(mockedRundownEventObserver.subscribeToRundownSetNext(anything())).once()
+  })
+
+  it('should subscribe to RundownReset changes upon init', async () => {
+    const mockedMiniShelfStateService: MiniShelfStateService = createMockOfMiniShelfStateService()
+    const mockedRundownTimingContextStateService: RundownTimingContextStateService = createMockOfRundownTimingContextStateService()
+    const mockedActionStateService: ActionStateService = createMockOfActionStateService()
+    const mockedRundownEventObserver: RundownEventObserver = createMockOfRundownEventObserver()
+    const mockedRundown: Rundown = new TestEntityFactory().createRundown()
```

Comment:
I would be very mindfull of that since a little change to the "default mock" could potentially break a lot of tests because you lose the overview of what part of the mock is being used for which test.

Reply:
And when writing new tests you would be drawn to alter the "default mock" since that is where you mocks are and every test needs a slightly different mock setup. So in reality developers would constantly be updating the "default mock" requiring them to always be aware of all the test in the suite using that mock

Reply:
If you alter the default mock in a way that breaks tests, then you should get quick feedback to indicate it. I have no strong feelings about it, so if this is disregarded, then fine :D